avi_vantage/datadog_checks/avi_vantage/check.py

```diff
@@ -0,0 +1,101 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from contextlib import contextmanager
+from copy import deepcopy
+
+from datadog_checks.avi_vantage import metrics
+from datadog_checks.base import AgentCheck, OpenMetricsBaseCheckV2
+from datadog_checks.base.errors import CheckException
+
+from .config_models import ConfigMixin
+
+RESOURCE_METRICS = {
+    'virtualservice': metrics.VIRTUAL_SERVICE_METRICS,
+    'pool': metrics.POOL_METRICS,
+    'controller': metrics.CONTROLLER_METRICS,
+    'serviceengine': metrics.SERVICE_ENGINE_METRICS,
+}
+LABELS_REMAPPER = {'type': 'avi_type', 'tenant': 'avi_tenant'}
+
+
+class AviVantageCheck(OpenMetricsBaseCheckV2, ConfigMixin):
+    __NAMESPACE__ = "avi_vantage"
+
+    def __init__(self, name, init_config, instances):
+        super(AviVantageCheck, self).__init__(name, init_config, instances)
+        # Required for storing the auth cookie
+        self.instance['persist_connections'] = True
+        self._base_url = None
+
+    @property
+    def base_url(self):
+        if not self._base_url:
+            self._base_url = self.config.avi_controller_url.rstrip('/')
+        return self._base_url
+
+    def configure_scrapers(self):
+        scrapers = {}
+
+        for entity in self.config.entities:
+            if entity not in RESOURCE_METRICS:
+                raise CheckException(
+                    f"Entity {entity} is not valid. Must be one of {', '.join(RESOURCE_METRICS.keys())}."
+                )
```

Comment:
Can this be checked in `avi_vantage/datadog_checks/avi_vantage/config_models/`?

Reply:
This is validated automatically already, I'm leaving the extra check here for double safety in case something change sin the future and/or config validation is disabled for some reason